### DISCUSSION TITLE
Is that __on_connection() thread only print the connection information and does not handle the connection as a separete thread

### DISCUSSION THREAD

**Discussion by dungdamquoc1993** [Y4hL/PyDoor#16] [General]
As I saw on this program there is 3 thread 
main thread
_accept thread (listening for incoming connection)
__on_connection (only use for print connection)
and the connections are not executed in separate threads is that right? 
Why shouldn't we create new thread for each connection

**Comment by tidely** [comment]
I tried this before and it just turned out to be very inefficient and not scalable. Since we are only ever interacting with one client at a time, having a thread for each connevtion is pointless. 

The 2 seperate threads that exist now to handle incoming connections are to seperate API calls and GUI. basically you should be able to call methods to interact with a connection without being forced to print stuff out to the terminal on connection.
